astropy/io/fits/column.py

```diff
@@ -1389,19 +1389,21 @@ def __getattr__(self, name):
 
         Implements for example self.units, self.formats, etc.
         """
-
         cname = name[:-1]
         if cname in KEYWORD_ATTRIBUTES and name[-1] == 's':
             attr = []
-            for col in self:
+            for col in self.columns:
                 val = getattr(col, cname)
-                if val is not None:
-                    attr.append(val)
-                else:
-                    attr.append('')
+                attr.append(val if val is not None else '')
             return attr
         raise AttributeError(name)
 
+    def _update_attrs(self, name=None):
+        """Get or update local lists of column attributes"""
+        attrs = [name] if name is not None else KEYWORD_ATTRIBUTES
+        for name in attrs:
+            setattr(self, name + 's', self.__getattr__(name + 's'))
```

Comment:
Specifically for the case of accessing a column by name I think that could probably be sped up without actually changing much of anything in `ColDefs`.  I'd have to think about it though.  Most other cases don't warrant this change though.  If it were just something that focused on faster lookup of columns by name I'd be for that.